test/events/events.go

```diff
@@ -35,19 +35,6 @@ func CheckEventsOrdered(t *testing.T, eventChan chan string, testName string, wa
 	return nil
```

Comment:
NIT: It's a bit inconsistent now that the unordered check has been moved inside to the fake client but the ordered one is in this test module. It might we worth keeping this logic in the same place, for maintenance and to help test developers. BTW, do you think the ordered check suffers from the same flake as the other one?

Reply:
This is something we could fix as a follow-up.

Reply:
Yes I'm looking into the k8s event as well, but I don't think they are the same root cause. 
Some prs:
https://github.com/tektoncd/pipeline/pull/5752
https://github.com/tektoncd/pipeline/pull/5770